posts/2020-11-20-JNDI-gRPC-200012.adoc

```diff
@@ -0,0 +1,389 @@
+
+[source, java]
+----
+package com.ibm.ws.grpc;
+
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+...
+@WebServlet(name = "grpcClient", urlPatterns = { "/grpcClient" }, loadOnStartup = 1)
+public class GrpcClientServlet extends HttpServlet {
+
+        ManagedChannel channel;
+        private GreeterGrpc.GreeterBlockingStub greetingService;
+
+        private void startService(String address, int port) 
+        {
+            channel = ManagedChannelBuilder.forAddress(address , port).usePlaintext().build();
+            greetingService = GreeterGrpc.newBlockingStub(channel);
+        }
+
+        private void stopService() 
+        {
+            channel.shutdownNow();
+        }
+
+        @Override
+        protected void doGet(HttpServletRequest reqest, HttpServletResponse response) 
+            throws ServletException, IOException 
+        {
+
+            // set user, address, port params
+        }
+
+        @Override
+        protected void doPost(HttpServletRequest request, HttpServletResponse response) 
+            throws ServletException, IOException 
+        {
+
+        // grab user, address, port params
+        startService(address, port);
+        HelloRequest person = HelloRequest.newBuilder().setName(user).build();
+        HelloReply greeting = greetingService.sayHello(person);
+
+        // send the greeting in a response
+        stopService();
+        }	
+    }
+}
+----
+
+As with the service example, the application must provide the link:https://github.com/grpc/grpc-java/blob/master/examples/src/main/proto/helloworld.proto[helloworld protof definition] along with the protobuf compiler output. All required gRPC client libraries are provided by the `grpcClient-1.0` feature.
+
+[#jndi]
+== Support for Custom JNDI Names for Enterprise Beans
+
+Support for Custom JNDI Names for Enterprise Beans (EJBs) is an enhancement to all existing enterprise beans features that allows an application to configure and use custom JDNI names for looking up and/or injecting enterprise beans, or use legacy default JNDI names instead of the specification defined JNDI names.
+
+Prior to this enhancement, Liberty only supported looking up enterprise beans using the specification defined JNDI names : `java:global/<app>/<module>/<bean>!<interface>` & variations for `java:app` and `java:module`.
+
+Without any additional configuration, legacy default JNDI names are now available for applications to use to lookup and/or inject enterprise beans. Also, rather than using the defaults, a custom name for each EJB may be specified in the `ibm-ejb-jar-bnd.xml` file (or `ibm-ejb-jar-bnd.xmi` file for EJB 2.x and 1.x modules). These new JNDI name options are available in addition to the existing specification required names.
```

Comment:
Combine this into one paragraph:
Prior to this enhancement, Liberty only supported looking up enterprise beans using the specification defined JNDI names : `java:global/<app>/<module>/<bean>!<interface>` & variations for `java:app` and `java:module`.  With this enhancement, and without any additional configuration, legacy default JNDI names are now available for use by applications to lookup and/or inject enterprise beans. Also, rather than using the defaults, a custom name for each EJB may be specified in the `ibm-ejb-jar-bnd.xml` file (or `ibm-ejb-jar-bnd.xmi` file for EJB 2.x and 1.x modules). These new JNDI name options are available in addition to the existing specification required names.

Reply:
Done